### ISSUE TITLE
put more effort over 'easy to use'

### ISSUE THREAD

**Issue by lilongyue** [apache/incubator-mxnet#18256]
After playing with a lot of projects implemented on 'mxnet' or 'pytorch'. 
the Project code length(python version) is often much shorter in Pytorch.

The main design pattern in mxnet is 'hybirdize module way'  , I feel we should put this behind the stage.
Let the user only deal with 'pythonic not symbolic' way of writing code . 

maybe we  should stop and reorganize  everything ?

**Comment by leezu** [comment]
I agree that many use-cases can be covered in a simpler fashion without hybridization. One question here is if "unhybridized mxnet" may be slower than "hybridized mxnet" and only the latter competitive with pytorch. I think @sxjscience is preparing some benchmarks.

**Comment by sxjscience** [comment]
@leezu Previously, we have one benchmark by @Jerryzcn : https://github.com/apache/incubator-mxnet/issues/17335 . According to my recent coding experience with the new DeepNumpy interface of MXNet, I do recognize that the usability of Gluon HybridBlock is worse than the Module of PyTorch. When implementing some slightly more advanced networks like TransformerXL, it's easy to write a PyTorch-based version that can fully utilize the GPU. On the other hand, it is not so trivial to do so in MXNet. I'm still trying to figure out the cause.
